File: leet_easy_c++/leet_189/leet_189a.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int len = nums.size();
        vector<int> nums2 = nums;
        for(int i=k, j=0; j<len; i++,j++) nums2[i%len]=nums[j];
        nums=nums2;
    }
};

File: leet_easy_javascript/leet_189/leet_189a.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    var res = [];
    res.length = nums.length;
    for(var i=0; i<nums.length; i++) res[(i+k)%nums.length]=nums[i];
    for(i=0; i<nums.length; i++) nums[i] = res[i];
};